### PR TITLE
Consolidated the group member status read funcs into an enum

### PR DESCRIPTION
**Pull request by mpretty-cyro**

### CODE REVIEW

File: include/session/config/groups/members.h [hunk 1]
```diff
@@ -11,6 +11,25 @@ extern "C" {
...
 
+/// A convenience status enum for the group member, this enum is returned by a function which
+/// reviews the various member status values (eg. invite, promotion, etc.) and returns a single
+/// consolidated value
+typedef enum GROUP_MEMBER_STATUS {
+    GROUP_MEMBER_STATUS_INVITE_UNKNOWN = 0,
+    GROUP_MEMBER_STATUS_INVITE_NOT_SENT = 1,
+    GROUP_MEMBER_STATUS_INVITE_FAILED = 2,
+    GROUP_MEMBER_STATUS_INVITE_SENT = 3,
+    GROUP_MEMBER_STATUS_INVITE_ACCPETED = 4,
```
Comment (Bilb) [inline include/session/config/groups/members.h]: typo here ACCPETED

File: include/session/config/groups/members.h [hunk 2]
```diff
@@ -11,6 +11,25 @@ extern "C" {
...
+    GROUP_MEMBER_STATUS_PROMOTION_UNKNOWN = 5,
+    GROUP_MEMBER_STATUS_PROMOTION_NOT_SENT = 6,
+    GROUP_MEMBER_STATUS_PROMOTION_FAILED = 7,
+    GROUP_MEMBER_STATUS_PROMOTION_SENT = 8,
+    GROUP_MEMBER_STATUS_PROMOTION_ACCPETED = 9,
```
Comment (Bilb) [inline include/session/config/groups/members.h]: same

**Review by Bilb — CHANGES_REQUESTED** [review]

File: include/session/config/groups/members.h (hunk 1 above)
Comment (mpretty-cyro) [reply include/session/config/groups/members.h]: Fixed

File: include/session/config/groups/members.h (hunk 2 above)
Comment (mpretty-cyro) [reply include/session/config/groups/members.h]: Fixed

**Review by Bilb — APPROVED** [review]

**Pull request merged**